components/script/dom/htmlinputelement.rs

```diff
@@ -1103,6 +1075,39 @@ impl HTMLInputElement {
     fn selection(&self) -> TextControlSelection<Self> {
         TextControlSelection::new(&self, &self.textinput)
     }
+
+    fn update_text_contents(&self, value: DOMString, update_text_cursor: bool) -> ErrorResult {
+        match self.value_mode() {
+            ValueMode::Value => {
+                // Steps 1-2.
+                let old_value = mem::replace(self.textinput.borrow_mut().single_line_content_mut(), value);
+                // Step 3.
+                self.value_dirty.set(true);
+                // Step 4.
+                self.sanitize_value(update_text_cursor);
+                // Step 5.
+                if *self.textinput.borrow().single_line_content() != old_value {
+                    self.textinput.borrow_mut().clear_selection_to_limit(Direction::Forward);
```

Comment:
This is the code that changes the edit point, I believe. We should add a check for update_text_cursor to the previous line.